backend/x86_proc.mli

```diff
@@ -107,7 +109,7 @@ end
 (** Support for plumbing a binary code emitter *)
 
 val internal_assembler :
-  ((Section_name.t * X86_ast.asm_program) list -> string -> unit) option ref
+  (X86_ast.asm_program ref Section_name.Tbl.t -> string -> unit) option ref
 
 val register_internal_assembler :
-  ((Section_name.t * X86_ast.asm_program) list -> string -> unit) -> unit
```

Comment:
It would be nice to define `internal_assembler` type and use it here and in `internal_assembler.mli`. 
```
type internal_assembler = -> string -> unit
```
Maybe also `type asm_code_by_section = X86_ast.asm_program ref Section_name.Tbl.t ` although we can't make it abstract because internal_assembler needs to iterate on it.  I mainly just don't like the "ref" part to be exposed.

Reply:
Unfortunately, X86_proc cannot depend on Internal_assembler because this creates a dependency cycle.